algorithmic_efficiency/workloads/criteo1tb/criteo1tb_jax/workload.py

```diff
@@ -0,0 +1,229 @@
+"""Criteo1TB DLRM-Small workload implemented in Jax."""
+import functools
+import math
+from typing import Dict, Optional, Tuple
+
+from clu import metrics as clu_metrics
+import flax
+from flax import jax_utils
+import jax
+import jax.numpy as jnp
+
+from algorithmic_efficiency import param_utils
+from algorithmic_efficiency import spec
+from algorithmic_efficiency.workloads.criteo1tb import input_pipeline
+from algorithmic_efficiency.workloads.criteo1tb.criteo1tb_jax import \
+    dlrm_small_model
+from algorithmic_efficiency.workloads.criteo1tb.criteo1tb_jax import metrics
+
+_NUM_DENSE_FEATURES = 13
+_VOCAB_SIZES = [1024 * 128] * 26
+
+
+# We use CLU metrics to handle aggregating per-example outputs across batches so
+# we can compute AUC metrics.
+@flax.struct.dataclass
+class EvalMetrics(clu_metrics.Collection):
+  loss: clu_metrics.Average.from_output("loss")
+  average_precision: metrics.BinaryMeanAveragePrecision
+  auc_roc: metrics.BinaryAUCROC
+
+
+class Criteo1TbDlrmSmallWorkload(spec.Workload):
+  """Criteo1TB DLRM-Small Jax workload."""
+
+  def __init__(self):
+    self._eval_iters = {}
+    self._param_shapes = None
+    self._param_types = None
+    self._flax_module = dlrm_small_model.DlrmSmall(
+        vocab_sizes=_VOCAB_SIZES,
+        total_vocab_sizes=sum(_VOCAB_SIZES),
+        num_dense_features=_NUM_DENSE_FEATURES)
+
+  def has_reached_goal(self, eval_result: float) -> bool:
+    return eval_result['validation/auc_roc'] > self.target_value
+
+  @property
+  def target_value(self):
+    return 0.8
+
+  @property
+  def loss_type(self):
+    return spec.LossType.SIGMOID_CROSS_ENTROPY
+
+  @property
+  def num_train_examples(self):
+    return 4_195_197_692
+
+  @property
+  def num_eval_train_examples(self):
+    return 100_000
+
+  @property
+  def num_validation_examples(self):
+    return 89_137_318  # DO NOT SUBMIT
+
+  @property
+  def num_test_examples(self):
+    return None
+
+  @property
+  def train_mean(self):
+    return 0.0
+
+  @property
+  def train_stddev(self):
+    return 1.0
+
+  @property
+  def max_allowed_runtime_sec(self):
+    return 8 * 60 * 60  # 8 hours. # DO NOT SUBMIT
```

Comment:
I believe, in our meeting, we decided to use 6 hours as the maximum runtime. But we might want to iterate on this anyway after seeing some experimental results.

Reply:
SGTM, done.